trust-quorum/src/task.rs

```diff
@@ -196,7 +414,10 @@ impl NodeTask {
                 Some(msg) = self.conn_mgr_rx.recv() => {
                     self.on_conn_msg(msg).await
                 }
+            }
 
+            for envelope in self.ctx.drain_envelopes() {
+                self.conn_mgr.send(envelope).await;
             }
```

Comment:
do we want to do this concurrently, or is serially okay? I guess this shouldn't be cancelled since there's an instruction to make `run` a top-level task.